clients/src/main/java/org/apache/kafka/common/protocol/ApiKeys.java

```diff
@@ -43,7 +44,12 @@
     DESCRIBE_GROUPS(15, "DescribeGroups"),
     LIST_GROUPS(16, "ListGroups"),
     SASL_HANDSHAKE(17, "SaslHandshake"),
-    API_VERSIONS(18, "ApiVersions"),
+    API_VERSIONS(18, "ApiVersions") {
+        @Override
+        public Struct parseResponse(short version, ByteBuffer buffer) {
+            return parseResponse(version, buffer, (short) 0);
```

Comment:
If client sends ApiVersionsRequest with a higher version that client supports, broker responds with a version 0 response that indicates unsupported version.